timely_beliefs/visualization/selectors.py

```diff
@@ -0,0 +1,195 @@
+)
+
+
+def horizon_selection_brush(init_belief_horizon=None) -> alt.MultiSelection:
+    """Create a brush for selecting one or multiple horizons.
+
+    :param init_belief_horizon: Optional initialisation value
+    """
+    if init_belief_horizon is None:
+        return alt.selection_multi(
+            nearest=False, encodings=["x"], empty="all", name="horizon_select"
+        )
+    else:
+        return alt.selection_multi(
+            nearest=False,
+            encodings=["x"],
+            name="horizon_select",
+            empty="all",
+            init={"belief_horizon": init_belief_horizon},
+        )
+
+
+def fixed_viewpoint_selector(
+    base, active_fixed_viewpoint_selector: bool = False
+) -> alt.Chart:
+    """Transparent selectors across the chart (visible on hover).
+    This is what tells us the belief time for a given x-value of the cursor.
+
+    :param active_fixed_viewpoint_selector: if False, return an idle colored version without tooltip
+    """
+    selector = base.mark_rule().encode(
+        x=alt.X("belief_time:T", scale={"domain": time_selection_brush.ref()}),
+        color=alt.ColorValue(idle_color)
+        if not active_fixed_viewpoint_selector
+        else alt.ColorValue("#c21431"),
+        opacity=alt.condition(nearest_x_hover_brush, alt.value(1), alt.value(0)),
+        tooltip=[
+            alt.Tooltip(
+                "belief_time:T",
+                timeUnit="yearmonthdatehoursminutes",
+                title="Click to select belief time",
+            )
+        ]
+        if active_fixed_viewpoint_selector is True
+        else None,
+    )
+    return selector.add_selection(nearest_x_select_brush).add_selection(
+        nearest_x_hover_brush
+    )
+
+
+def time_window_selector(base, interpolate: bool) -> alt.LayerChart:
+    if interpolate is True:
+        tws = base.mark_area(interpolate="monotone")
+    else:
+        tws = base.mark_bar().encode(x2=alt.X2("event_end:T"))
+    tws = tws.encode(
+        x=alt.X("event_start", title=""),
+        y=alt.Y(
+            "reference_value",
+            title="",
+            axis=alt.Axis(values=[], domain=False, ticks=False),
+        ),
+        color=alt.ColorValue(idle_color),
+        tooltip=alt.TooltipValue("Click and drag to select time window"),
+    ).properties(height=30, title="Select time window")
+    tws = tws.add_selection(time_selection_brush) + tws.transform_filter(
+        time_selection_brush
+    ).encode(
+        color=alt.condition(
+            time_selection_brush, alt.ColorValue("#c21431"), alt.ColorValue(idle_color)
+        )
+    )
+    return tws
+
+
+def horizon_selector(
+    base,
+    horizon_selection_brush: alt.MultiSelection,
+    belief_horizon_unit: str,
+    intuitive_forecast_horizon: bool,
+    unique_belief_horizons,
+) -> alt.LayerChart:
+    bar_chart = (
+        base.mark_rule(orient="vertical")
+        # # Remove the most recent belief horizon from the selector
```

Comment:
Can these lines go?

Reply:
Yes, they just reminded me to finish moving from belief horizon to forecast horizon.